api_handlers.go

```diff
@@ -296,3 +310,50 @@ func apiDiffHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	w.Write(bytes)
 }
+
+// handleAPIDiffPost handles POST requests to /api/diff, which allows the caller to produce the diff of an arbitrary
+// run result JSON blob against a historical production run.
+func handleAPIDiffPost(w http.ResponseWriter, r *http.Request) {
```

Comment:
I actually found this quite readable, even with all the error handling. Maybe Go's OK :)